app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -376,6 +378,25 @@ class DefaultSessionControlController(
             metrics.track(Event.PocketTopSiteClicked)
         }
 
+        val baseURL = url.tryGetHostFromUrl()
+        if (SupportUtils.GOOGLE_URL.contains(baseURL)) {
+            val availableEngines = activity
+                .components
+                .core
+                .store
+                .state
+                .search
+                .searchEngines
+
+            val searchAccessPoint = Event.PerformedSearch.SearchAccessPoint.TOPSITE
+            availableEngines.firstOrNull { engine -> engine.suggestUrl?.contains(baseURL) == true }
+                ?.let { searchEngine ->
+                    searchAccessPoint.let { sap ->
+                        MetricsUtils.createSearchEvent(searchEngine, store, sap)
```

Comment:
I am not super familiar with this code, but I want to make sure that this is implemented in a way where this event is fired when a search is done and not when the user taps on the tile. A tap may not always lead to a search.

The definitive point where to send this event is when the search is submitted to the search engine.

Reply:
Yes, I agree. I think our tracking of this should happen in the `SearchDialogController` where we are checking the other `searchAccessPoint`s: https://github.com/mozilla-mobile/fenix/blob/acae32095038ffef7569152aa43c5650bdac72fe/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt#L98